validator/testcases/javascript/actions.py

```diff
@@ -81,16 +78,26 @@ def trace_member(traverser, node, instantiate=False):
         base = trace_member(traverser, node["object"], instantiate)
         base = _expand_globals(traverser, base)
 
+        identifier = _get_member_exp_property(traverser, node)
+
         # Handle the various global entity properties.
         if base.is_global:
             # If we've got an XPCOM wildcard, return a copy of the entity.
             if "xpcom_wildcard" in base.value:
                 traverser._debug("MEMBER_EXP>>XPCOM_WILDCARD")
+
+                from predefinedentities import CONTRACT_ENTITIES
+                if identifier in CONTRACT_ENTITIES:
+                    kw = dict(err_id=("js", "actions", "dangerous_contract"),
+                              warning="Dangerous XPCOM contract ID")
```

Comment:
On Tue, May 05, 2015 at 07:46:49AM -0700, Mathieu Agopian wrote:
>>          # Handle the various global entity properties.
>>          if base.is_global:
>>              # If we've got an XPCOM wildcard, return a copy of the entity.
>>              if "xpcom_wildcard" in base.value:
>>                  traverser._debug("MEMBER_EXP>>XPCOM_WILDCARD")
>> +
>> +                from predefinedentities import CONTRACT_ENTITIES
>> +                if identifier in CONTRACT_ENTITIES:
>> +                    kw = dict(err_id=("js", "actions", "dangerous_contract"),
>> +                              warning="Dangerous XPCOM contract ID")
>
>Why not define all the dict items in the same place (either here or in `prefedinedentities`)?

Well, the main reason is that I don't want to have to have to 
duplicate keys.

At the moment, all of the contract IDs in that dict have the 
same properties, so that's not an issue. However, it's always 
worried me that we tend not to do checks for contract IDs, only 
interface names, so I intend to add more in the future.

The slightly more complicated reason is that most of the 
definitions in `predefinedentities` follow a similar pattern, 
where they can accept a string, for only the description, a 
dict, which can define other properties as well, or a callable 
which... has completely different behavior depending on exactly 
where you use it.

All of this is extremely messy and inconsistent, so what I'd 
really like to do is unify the handling of all of them via some 
kind of helper function, which means any new definitions along 
these lines need to follow the same behavior in any case.